pkg/api/findings.go

```diff
@@ -41,6 +44,70 @@ func (api *API) GetFinding(c echo.Context) error {
 	return response(c, finding, "finding")
 }
 
+// PatchFinding updates parts of a finding and returns a
+// JSON containing the modified finding.
+func (api *API) PatchFinding(c echo.Context) error {
+	id := c.Param("id")
+
+	finding, err := api.storage.GetFinding(id)
+	if err != nil {
+		return err
+	}
+
+	if finding.ID == "" {
+		return echo.NewHTTPError(http.StatusNotFound)
+	}
+
+	payload := new(model.PatchFindingPayload)
+	if err := c.Bind(payload); err != nil {
+		return err
+	}
+
+	if payload.Status != "" {
+		if !validFindingStatus(payload.Status) {
+			return echo.NewHTTPError(http.StatusUnprocessableEntity, fmt.Sprintf("Invalid status: '%s'", payload.Status))
+		}
+
+		// Valid transitions:
+		//
+		// OPEN           -> OPEN
+		// FALSE_POSITIVE -> OPEN
+		// OPEN           -> FALSE_POSITIVE
+		// FALSE_POSITIVE -> FALSE_POSITIVE
+		if finding.Status != "OPEN" && finding.Status != "FALSE_POSITIVE" {
+			return echo.NewHTTPError(http.StatusConflict, fmt.Sprintf("Status transition not allowed: from '%s' to '%s'", finding.Status, payload.Status))
+		}
+
+		if payload.Status != "OPEN" && payload.Status != "FALSE_POSITIVE" {
+			return echo.NewHTTPError(http.StatusConflict, fmt.Sprintf("Status transition not allowed: from '%s' to '%s'", finding.Status, payload.Status))
+		}
```

Comment:
Should we merge these two if statements into one with an OR? And possibly move the transition validation to a separate function?

Also, I think we can avoid the `validFindingsStatus` function if we already validate the transitions explicitly.

Reply:
Done. Merged the two IF statements and moved them to a separate function, named `validateFindingTransition`.

Regarding the avoidance of `validFindingsStatus`, I am not sure if I do agree/understand. The `validFindingsStatus` ensures that the user provides a valid status and not something like `{"status": "foo"}`. The transition validation is not taking this case into account. What do you think? Is it acceptable to keep the `validFindingsStatus` code?

Reply:
I see your point, what you mean is that if we don't validate the status individually then we might return an error message like: (following your example)
`Status transition not allowed: from 'OPEN' to 'foo'` 
Right?

Well I see that as a possible consideration.
On the other hand having a map with the possible statuses makes us modify this map if we add new statuses etc.

No big deal from my part, I will let you decide which option to use :+1:

Reply:
Exactly. If the user submits `status:"foo"`, then I want to explicitly tell him that he submitted an invalid status type, like:
`Invalid status: 'foo'`

Instead telling him that he tried an invalid status transition:
`Status transition not allowed: from 'OPEN' to 'foo'`

Also, keeping the map of possible statuses will prevent us from forgetting to handle corner cases in a future scenario.